discovery-infra/test_infra/controllers/node_controllers/host.py

```diff
@@ -0,0 +1,68 @@
+import logging
+from test_infra.controllers.node_controllers import ssh
+
+
+class Host:
+
+    def __init__(self, name, node_controller, private_ssh_key_path=None, username="core"):
+        self.name = name
+        self.private_ssh_key_path = private_ssh_key_path
+        self.username = username
+        self.node_controller = node_controller
+        self._ips = []
+        self._macs = []
+
+    @property
+    def is_active(self):
+        return self.node_controller.is_active(self.name)
+
+    def _set_ips_and_macs(self):
+        self._ips, self._macs = self.node_controller.get_node_ips_and_macs(self.name)
+
+    # TODO maybe add ttl? need mechanism that
+    #  will zero this value when node is stopped
+    @property
+    def ips(self):
+        if not self._ips:
+            self._set_ips_and_macs()
+        return self._ips
+
+    @property
+    def macs(self):
+        if not self._macs:
```

Comment:
Maybe instead of this check everytime we should just run a `refresh` function?

for example
```python
    def refresh(self):
        self._ips, self._macs = self.node_controller.get_node_ips_and_macs(self.name)

  @property
    def ips(self):
        self.refresh()
        return self._ips

```

Reply:
not sure how it will help? the idea was not to run it everytime

Reply:
The question is why it should be changed at all.
If it isn't supposed to - you can refresh only after `start` or what ever, but if you check every time you call to `ips` it means that you think it might have changed (iiuc), so you should refresh each time.

Reply:
i think we can start this way, it is relatively easy and must not change till test ends

Reply:
when we will move to use start though host class, we can move to update it there.